test/OpenBCISample-test.js

```diff
@@ -30,76 +30,79 @@ var sampleBuf = samplePacket();
 
 
 describe('openBCISample',function() {
-    describe('#convertPacketToSample', function() {
-        it('should have the correct start byte', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(k.OBCIByteStart,sample.startByte);
-        });
-        it('should have the correct stop byte', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(k.OBCIByteStop,sample.stopByte);
+    var channelScaleFactor = 4.5 / 24 / (Math.pow(2,23) - 1);
+    describe('#parseRawPacket', function() {
+        it('should fulfill promise', function() {
+            return openBCISample.parseRawPacket(sampleBuf).should.be.fulfilled;
         });
         it('should have the correct sample number', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(0x45,sample.sampleNumber);
-        });
-        it('all the channels should have the same number value as their (index + 1) * scaleFactor', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            for(var i = 0;i < k.OBCINumberOfChannelsDefault;i++) {
-                //console.log(openBCISample.scaleFactorChannel * i);
-                assert.equal(sample.channelData[i],openBCISample.scaleFactorChannel * (i+1));
-            }
-        });
-        it('all the auxs should have the same number value as their index * scaleFactor', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            for(var i = 0;i < 3;i++) {
-                assert.equal(sample.auxData[i],openBCISample.scaleFactorAux * i);
-            }
-        });
-        it('check to see if negative numbers work on channel data',function() {
+            return openBCISample.parseRawPacket(sampleBuf).should.eventually.have.property('sampleNumber').equal(0x45);
+        });
+        it('all the channels should have the same number value as their (index + 1) * scaleFactor', function(done) {
```

Comment:
sampleBuf has it's channel number for each 3 byte integer. See line 20. So parse the sample we created and each value resulting from the channelData array should be it's index + 1 (i.e. channel number) multiplied by the channel scale factor set by the ADS1299 for a gain of 24 (default)